include/rmm/device_scalar.hpp

```diff
@@ -82,6 +82,25 @@ class device_scalar {
     set_value(initial_value, stream);
   }
 
+  /**
+   * @brief Construct a new `device_scalar` by deep copying the contents of
+   * another `device_scalar`, optionally using the specified stream and memory
```

Comment:
The stream and memory resource are always used, they just have defaults.

```suggestion
   * another `device_scalar`, using the specified stream and memory
```